### PR TITLE
[bitnami/mongodb] add existingSecret for tls certificates

### PR DESCRIPTION
**Pull request by benjamin-wright**
**Description of the change**

Adding a `tls.existingSecret` option to allow existing certificates to be passed in to the mongo cluster

**Benefits**

Easier to use custom certs, rather than relying on the certs generated in the helm chart

**Possible drawbacks**

Backwards compatibile, so no obvious issues beyond the minor increase in complexity

**Applicable issues**

NA - can raise one if there's some discussion needed?

**Additional information**

NA

**Checklist** 
- [X] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [X] Variables are documented in the README.md
- [X] Title of the PR starts with chart name (e.g. `[bitnami/chart]`)
- [X] If the chart contains a `values-production.yaml` apart from `values.yaml`, ensure that you implement the changes in both files

### CODE REVIEW

**Comment by deveshk0** [comment]
@benjamin-wright Sorry didn't see your PR, I have added a similar one to use custom ca

**Comment by benjamin-wright** [comment]
> @benjamin-wright Sorry didn't see your PR, I have added a similar one to use custom ca https://github.com/bitnami/charts/pull/4690

@deveshk0 No worries, great minds think alike! Both look like orthogonal improvements I think, the existing secret approach is useful for me with sealed secrets, but i can definitely see the benefit of yours for terraform / helm install / etc.

File: bitnami/mongodb/templates/replicaset/statefulset.yaml [hunk 1]
```diff
@@ -450,7 +450,7 @@ spec:
           emptyDir: {}
         - name: certs-volume
           secret:
-            secretName: ca-clients
+            secretName: {{ default "ca-clients" .Values.tls.existingSecret }}
```
Comment (juan131) [inline bitnami/mongodb/templates/replicaset/statefulset.yaml:453]: I just noticed the default secret is hardcoded to "ca-clients", which is clearly wrong. We should use sth like:'
```
  name: {{ include "mongodb.fullname" . }}-metrics
```

Could you please fix it both here and the **secrets-ca.yaml** file (line 9)?

File: bitnami/mongodb/Chart.yaml [hunk 2]
```diff
@@ -26,4 +26,4 @@ name: mongodb
 sources:
   - https://github.com/bitnami/bitnami-docker-mongodb
   - https://mongodb.org
-version: 10.1.4
+version: 10.1.5
```
Comment (juan131) [inline bitnami/mongodb/Chart.yaml:29]: Please bump minor version instead since you're adding a new parameter.

**Review by juan131 — CHANGES_REQUESTED** [review]
Thanks for the contribution!! 

I agree with @benjamin-wright that both this and the solution provided @deveshk0 are valid ones. They are different approaches but I don't see any problem on supporting both.

Please review my comments when you have a chance.

File: bitnami/mongodb/Chart.yaml (hunk 2 above)
Comment (benjamin-wright) [reply bitnami/mongodb/Chart.yaml]: Done

File: bitnami/mongodb/templates/replicaset/statefulset.yaml (hunk 1 above)
Comment (benjamin-wright) [reply bitnami/mongodb/templates/replicaset/statefulset.yaml]: @juan131 updated, does this look ok? Pulling it out into a secret seemed like the easiest solution

**Review by juan131 — APPROVED** [review]
Great! Your PR looks perfect now! Approving it

**Pull request merged**